packages/transaction-controller/src/TransactionController.ts

```diff
@@ -3085,6 +3104,46 @@ export class TransactionController extends BaseController<
     return txsToKeep;
   }
 
+  /**
+   * Trim the amount of transactions that are set on the state. Checks if the
+   * length of the tx history is longer then desired persistence limit and then
+   * if it is removes the oldest confirmed or rejected tx. For safety of
+   * presenting a fully functional transaction UI representation, this function
+   * will not break apart transactions created on the same day, per network. Not
+   * accounting for transactions of the same nonce, same day and network combo
+   * can result in confusing or broken experiences in the UI.
+   *
+   * @param transactions - The transactions to be applied to the state.
+   * @returns The trimmed list of transactions.
+   */
+  #trimBatchTransactionsForState(
```

Comment:
Can this also go in `batch.ts` file for the sake for modularity?

Plus can we simplify this to just add at the end, and delete from the start once we exceed some limit?

Reply:
Removed `#trimBatchTransactionsForState`, becasue we are not saving the `transactionBatches` yet